soc/nordic/Kconfig

```diff
@@ -199,7 +199,7 @@ config NRF_PLATFORM_HALTIUM
 
 config DEPRECATED_UART_NRFX_UARTE_LEGACY_SHIM
 	bool
-	default y if !UART_NRFX_UARTE_LEGACY_SHIM
+	default y if !UART_NRFX_UARTE_LEGACY_SHIM && UART_NRFX_UARTE && UART_NRFX && SERIAL
```

Comment:
isn't that enough?
```suggestion
	default y if !UART_NRFX_UARTE_LEGACY_SHIM && UART_NRFX_UARTE
```

Reply:
Both should be addressed in latest version.